### DISCUSSION TITLE
How to use reflection/tsyringe with parcel 2

### DISCUSSION THREAD

**Discussion by K0IN** [parcel-bundler/parcel#5238] [Q&A] [answered]
I'm not able to use tsyringe due to missing metadata, my project works on parcel 1 with no problems tho. 
my code: is from https://github.com/microsoft/tsyringe#example-without-interfaces
[code](https://i.imgur.com/OQ6dv40.png)

Also, I think this is a NOT a bug more likely a wrong setting in babel?.
Bablerc:
`{
    "presets": ["@babel/preset-env", "@babel/preset-typescript"],
    "plugins": [
        ["@babel/plugin-proposal-decorators", { "legacy": true }],
        ["@babel/plugin-proposal-class-properties", { "loose" : true }],
        "@babel/plugin-transform-classes"
    ]
}`

parcelrc:
`{
    "extends": "@parcel/config-default"
}`

**Comment by mischnic** [comment, marked as answer]
You probably need to use tsc instead of Babel to transpile. Here's how to configure Parcel 2 to do that:  https://v2.parceljs.org/getting-started/migration/#babel-doesn't-read-tsconfig.json

**Reply by K0IN** [reply]
Still not working...
Tried: https://i.imgur.com/fWn7X7t.png 
result: https://i.imgur.com/FHH7gEx.png
any idea?

**Reply by mischnic** [reply]
Works for me: https://github.com/mischnic/parcel-issue-5238-tsyringe
